### PR TITLE
Update "select a storage driver" with current status

### PR DESCRIPTION
**Pull request by thaJeztah**
The information on this page was a bit outdated, and with "overlay" and "devicemapper" being phased-out, could use some changes;

- de-emphasize "overlay", "devicemapper" and "aufs"
- split "recommended" and "alternative drivers" columns
  (vfs was listed as a "recommended" driver, but definitely
  is not recommended for production use).
- add notes about "overlay" and "devicemapper" being
  deprecated in Docker 18.09
- some other textual changes.

### Related issues (optional)

- https://github.com/docker/cli/pull/1424 Deprecate "devicemapper" storage driver. (see https://github.com/docker/cli/pull/1424#issuecomment-437162893)
- https://github.com/docker/cli/pull/1425 Deprecate legacy overlay storage driver
- https://github.com/docker/cli/pull/1455 [18.09 backport] deprecate devicemapper and legacy overlay storage drivers

### CODE REVIEW

**Comment by thaJeztah** [comment]
ping @ahh-docker @bermudezmt  @mataneja PTAL

**Comment by thaJeztah** [comment]
oh, and @L-Hudson 🤗

**Comment by ahh-docker** [comment]
@thaJeztah Mind if I take some edits at this before sending them off to @L-Hudson for review?

File: storage/storagedriver/select-storage-driver.md [hunk 1]
```diff
@@ -26,19 +26,28 @@ this decision, there are three high-level factors to consider:
...
+      current versions of CentOS and RHEL now have support for `overlay2`, and
+      is now the recommended driver.
+    - The `btrfs` and `zfs` storage drivers are used if they are the backing
+      filesystem (the filesystem of the host on which Docker is installed).
+      These filesystems allow for advanced options, such as creating "snapshots",
+      but require more maintenance and setup. Each of these relies on the backing
+      filesystem being configured correctly.
+    - The `vfs` storage driver is intended for testing purposes, and in situations
+      where no copy-on-write filesystem can be used. Performance of this storage
+      driver is low, and because it provides no deduplication, will result in
```
Comment (AkihiroSuda) [inline storage/storagedriver/select-storage-driver.md]: It still dedupe when copyfilerange is available?

Comment (thaJeztah) [reply storage/storagedriver/select-storage-driver.md]: Oh, good one. Yeah, we can remove this bit (my main reason for mentioning it was to discourage "regular" users from using it; I know there's specific use-cases for using vfs, but for most users it should not something to look at)

**Comment by thaJeztah** [comment]
@ahh-docker no, not at all; edit away; I started working on this, then saw the discussion that the docs team may started working on it, so I thought: let me push, and open as a PR.

File: storage/storagedriver/select-storage-driver.md [hunk 2]
```diff
@@ -21,56 +21,54 @@ After you have read the [storage driver overview](index.md), the
...
-- Some storage drivers require you to use a specific format for the backing
-  filesystem. If you have external requirements to use a specific backing
-  filesystem, this may limit your choices. See
-  [Supported backing filesystems](#supported-backing-filesystems).
-
-- After you have narrowed down which storage drivers you can choose from, your
-  choice are determined by the characteristics of your workload and the
-  level of stability you need. See [Other considerations](#other-considerations)
-  for help making the final decision.
+If multiple storage drivers are supported in your kernel, Docker has prioritized
```
Comment (thaJeztah) [inline storage/storagedriver/select-storage-driver.md]: dropped "a" here (Docker has prioritized -> Docker has a propritized)

File: storage/storagedriver/select-storage-driver.md [hunk 3]
```diff
@@ -21,56 +21,54 @@ After you have read the [storage driver overview](index.md), the
...
+
+* `overlay2` is the preferred storage driver, for all currently supported
+   Linux distributions, and requires no extra configuration.
+* `aufs` is the preferred storage driver for Docker 18.06 and older, when
+   running on Ubuntu 14.04 on kernel 3.13 (which has no support for `overlay2`.
+* `devicemapper` is supported, but requires `direct-lvm` for production
+   environments, because `loopback-lvm`, while zero-configuration, has very
+   poor performance. `devicemapper` was the recommended storage driver for
+   CentOS and RHEL, as their kernel version did not support `overlay2`. However,
+   current versions of CentOS and RHEL now have support for `overlay2`, and
```
Comment (thaJeztah) [inline storage/storagedriver/select-storage-driver.md]: Looks like my mistake; `, and is now the recommended driver` should probably be `, which is now the recommended driver`

File: storage/storagedriver/select-storage-driver.md (hunk 2 above)
Comment (ahh-docker) [reply storage/storagedriver/select-storage-driver.md]: fixed.

File: storage/storagedriver/select-storage-driver.md (hunk 3 above)
Comment (ahh-docker) [reply storage/storagedriver/select-storage-driver.md]: fixed

File: storage/storagedriver/select-storage-driver.md (hunk 1 above)
Comment (ahh-docker) [reply storage/storagedriver/select-storage-driver.md]: fixed

**Comment by ahh-docker** [comment]
Asking @bermudezmt or @ddeyo for peer review. Once I get a LGTM, I'll merge.

File: storage/storagedriver/select-storage-driver.md [hunk 4]
```diff
@@ -99,13 +98,13 @@ configurations work on recent versions of the Linux distribution:
 | Docker CE on CentOS | `overlay2`                                                             | `overlay`¹, `devicemapper`², `zfs`, `vfs`         |
 | Docker CE on Fedora | `overlay2`                                                             | `overlay`¹, `devicemapper`², `zfs`, `vfs`         |
 
-¹) The `overlay` storage driver is marked deprecated in docker 18.09, and will be
-removed in a future release. Users of the `overlay` storage driver are recommended
-to migrate to `overlay2`.
+¹) The `overlay` storage driver is deprecated in Docker 18.09, and will be
```
Comment (bermudezmt) [inline storage/storagedriver/select-storage-driver.md]: I'm a little confused by the usage of "Docker 18.09" here. Is it implicit that this is Engine or should we explicitly mention that?

**Comment by bermudezmt** [comment]
I spotted a few things that needed changing, and would like to get a review from @L-Hudson on the product naming especially around "Docker Engine Enterprise and Docker EE" and 'Docker Engine Community and Docker CE".

File: storage/storagedriver/select-storage-driver.md (hunk 4 above)
Comment (thaJeztah) [reply storage/storagedriver/select-storage-driver.md]: adding "Engine" won't hurt

Comment (ahh-docker) [reply storage/storagedriver/select-storage-driver.md]: s/Docker/Docker Engine - Enterprise Thanks!

**Comment by ahh-docker** [comment]
@L-Hudson: Please give your final review and edits. Once it's done, lemme know so we can merge. Please tag myself and @thaJeztah  with any questions. Thanks!

File: storage/storagedriver/select-storage-driver.md [hunk 5]
```diff
@@ -73,25 +78,34 @@ In addition, Docker does not recommend any configuration that requires you to
 disable security features of your operating system, such as the need to disable
 `selinux` if you use the `overlay` or `overlay2` driver on CentOS.
 
-### Docker EE and CS-Engine
+### Docker Engine Enterprise and Docker EE
```
Comment (ddeyo) [inline storage/storagedriver/select-storage-driver.md]: shd be "Docker Engine - Enterprise" and "Docker Enterprise"

File: storage/storagedriver/select-storage-driver.md [hunk 6]
```diff
@@ -73,25 +78,34 @@ In addition, Docker does not recommend any configuration that requires you to
...
 
-For Docker EE and CS-Engine, the definitive resource for which storage drivers
-are supported is the
+For Docker Engine Enterprise and Docker EE, the definitive resource for which
+storage drivers are supported is the
 [Product compatibility matrix](https://success.docker.com/Policies/Compatibility_Matrix).
 To get commercial support from Docker, you must use a supported configuration.
 
-### Docker CE
+### Docker Engine Community and Docker CE
```
Comment (ddeyo) [inline storage/storagedriver/select-storage-driver.md]: shd be "Docker Engine - Community"

File: storage/storagedriver/select-storage-driver.md [hunk 7]
```diff
@@ -73,25 +78,34 @@ In addition, Docker does not recommend any configuration that requires you to
...
 
-For Docker CE, only some configurations are tested, and your operating system's
-kernel may not support every storage driver. In general, the following
+For Docker Engine Community, only some configurations are tested, and your operating
```
Comment (ddeyo) [inline storage/storagedriver/select-storage-driver.md]: shd be "Docker Engine - Community"

File: storage/storagedriver/select-storage-driver.md (hunk 5 above)
Comment (ahh-docker) [reply storage/storagedriver/select-storage-driver.md]: fixed throughout.

File: storage/storagedriver/select-storage-driver.md (hunk 7 above)
Comment (ahh-docker) [reply storage/storagedriver/select-storage-driver.md]: Fixed throughout.

**Pull request merged**